packages/sdk/react-shell/src/panels/IdentityPanel/steps/AgentPanel.tsx

```diff
@@ -0,0 +1,161 @@
+
+  useEffect(() => {
+    const fetchAgentStatus = async () => {
+      // TODO(nf): if agent exists, subscribe to updates instead of oneshot status
+      const agentStatus = await agentHostingProviderClient?.getAgent(identity?.identityKey.truncate() ?? 'foo');
+      if (agentStatus) {
+        setAgentActive(true);
+        setAgentStatus(agentStatus);
+      } else {
+        setAgentActive(false);
+        setValidationMessage('No agent deployed via provider.');
+      }
+    };
+    void fetchAgentStatus();
+  }, []);
+
+  const onAgentCreateEvent = useCallback(async (invitation: Invitation) => {
+    const invitationCode = InvitationEncoder.encode(invitation);
+    if (invitation.state === Invitation.State.CONNECTING) {
+      console.log(JSON.stringify({ invitationCode, authCode: invitation.authCode }));
+      if (!identity) {
+        setValidationMessage('Identity not found?');
+        console.error('Identity not found');
+        return;
+      }
+      agentHostingProviderClient
+        ?.createAgent(invitationCode, identity.identityKey.truncate())
+        .then((res) => {
+          console.log(res);
+          if (res) {
+            setAgentStatus(res);
+            setAgentActive(true);
+            setValidationMessage('');
+          } else {
+            setValidationMessage('error creating agent');
+          }
+        })
+        .catch((error) => {
+          // TODO(nf): feedback on error
+          setValidationMessage('error creating agent');
+        });
+      // TODO(nf): feedback on response
+    }
+  }, []);
+
+  const handleAgentCreate = async () => {
+    invitations.forEach((invitation) => invitation.cancel());
+
+    // TODO(nf): do this work in the hosting provider client?
+    const invitation = client.halo.share({ type: Invitation.Type.MULTIUSE, authMethod: Invitation.AuthMethod.NONE });
+
+    invitation.subscribe(onAgentCreateEvent);
+  };
+
+  const handleAgentDestroy = async () => {
+    const identity = client.halo.identity.get();
+    if (!identity) {
+      console.error('Identity not found');
+      return;
+    }
+    agentHostingProviderClient
+      ?.destroyAgent(identity.identityKey.truncate())
+      .then((res) => {
+        if (res) {
+          setValidationMessage('No agent deployed via provider.');
+          setAgentStatus('');
+          setAgentActive(false);
+        } else {
+          // TODO(nf): feedback on error
+          setValidationMessage('error destroying agent.');
+        }
+      })
+      .catch((error) => {
+        setValidationMessage('error destroying agent.');
+      });
+  };
+
+  return (
+    <AgentFormImpl
+      {...props}
+      onAgentCreate={handleAgentCreate}
+      onAgentDestroy={handleAgentDestroy}
+      validationMessage={validationMessage}
+      agentStatus={agentStatus}
+      agentActive={agentActive}
+    />
+  );
+};
+
+// TODO(zhenyasav): impl shouldn't need send()
+export const AgentFormImpl = (props: AgentFormImplProps) => {
```

Comment:
Impl is a bit strange; why do you need another function?
Also, generally destructure inside function: 
e.g., `const AgentForm = ({ ... }: Props) => `